src/Input.js

```diff
@@ -6,23 +6,36 @@ import uuid from 'node-uuid';
 
 class Input extends React.Component {
   componentDidMount() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      $(ReactDOM.findDOMNode(this.refs.inputEl)).material_select();
-    }
+    this._setupEl();
   }
-
   componentDidUpdate() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      var $el = $(ReactDOM.findDOMNode(this.refs.inputEl));
-      $el.material_select('destroy');
-      $el.material_select();
+    this._setupEl();
+  }
+
+  _setupEl() {
+    if (typeof $ !== 'undefined') {
+      var $inputEl = $(ReactDOM.findDOMNode(this.refs.inputEl));
+
+      if (this.props.type === 'select' && !this.props.browserDefault) {
+        $inputEl.material_select('destroy');
+        $inputEl.material_select();
+      }
+      else {
+        var $labelEl = $(ReactDOM.findDOMNode(this.refs.labelEl));
+        if ($inputEl.val()) {
+          $labelEl.addClass('active');
```

Comment:
This is not the best way to work with react, you should / can not modify DOM outside of react, for dynamic class names, always use the `classnames` function.

```javascript
import cx from 'classnames';

constructor(props) {
  super(this);
  this.state = { active: this.props.defaultValue !== null };
  this.onChange = this.onChange.bind(this);
}
render() {
  let labelClasses = {
     active: this.state.active,
  };
  let htmlLabel = <label className={cx(labelClasses)} htmlFor={id}>{label}</label>;

  // Listen on the change event of the input
  <C onChange={this.onChange} {...props} />
}
onChange(e) { this.state.active = $inputEl.val() !== null; }
```